lib/wallet/coinselector.js

```diff
@@ -0,0 +1,882 @@
+        // if this CoinPointer's value is different from the previous one
+        if (   selected.length === 0
+            || index - 1 === selected[selected.length - 1]
+            || pointer.effectiveValue !==
+               this.coinPointers[index - 1].effectiveValue
+        ) {
+          selected.push(index);
+          currValue += this.coinPointers[index].effectiveValue;
+        }
+      }
+      index++;
+    }
+
+    return bestSelection;
+  }
+
+  /**
+   * Initiate selection using Lowest Larger selection algorithm.
+   * @param {Number} target - Selection target
+   * @returns {Number[]} selected - array of indicies of selected coins
+   */
+
+  selectLowestLarger(target) {
+    // while target is greater than
+    // the largest coin we have, we
+    // will keep selecting the largest coin
+
+    let index = 0;
+    const selected = [];
+    let effectiveValue = this.coinPointers[index].effectiveValue;
+    while (target >= effectiveValue) {
+      // update target, select current coin and increment index
+      selected.push(index);
+      target -= effectiveValue;
+      index++;
+
+      if (index === this.coinPointers.length)
+        break;
+
+      effectiveValue = this.coinPointers[index].effectiveValue;
+    }
+
+    if (target > 0 && index !== this.coinPointers.length) {
+      // now we are sure that target < largest unselected coin
+      // we will perform Binary search to find the smallest coin
+      // which is greater than target value
+
+      const lowestLargerIndex = this.findLowestLarger(target, index);
+      target -= this.coinPointers[lowestLargerIndex].effectiveValue;
+      selected.push(lowestLargerIndex);
+    }
+
+    return target > 0 ? [] : selected;
+  }
+
+  /**
+   * Find smallest coin greater than
+   * the target using binary search
+   * @param {Number} target
+   * @returns {Number} index
+   */
+
+  findLowestLarger(target, index) {
+    let i = index;
+    let j = this.coinPointers.length - 1;
+
+    // begin binary search
+    let lowestLargerIndex = 0;
+    let mid = 0;
+    while (i <= j) {
+      mid = Math.floor((i + j) / 2);
+
+      // calculate effective value of coin at mid
+      const effectiveValue = this.coinPointers[mid].effectiveValue;
+
+      // if target is less than coin at mid
+      // then search in right part of array
+      if (target <= effectiveValue) {
+        lowestLargerIndex = mid;
+        // repeat for right half
+        i = mid + 1;
+      } else {
+        // repeat for left half
+        j = mid - 1;
+      }
+    }
+
+    return lowestLargerIndex;
+  }
+
+  /**
+   * Calculate effectiveValue of a coin
+   * effectiveValue = coin.value - cost of spending that coin
+   * @param {Coin} coin - A coin object
+   * @returns {Number} - effective value of the coin
+   */
+
+  async getEffectiveValue(coin) {
+    const spendingSize = await coin.estimateSpendingSize(this.getAccount);
+    return coin.value - this.getFee(spendingSize);
+  }
```

Comment:
I don't think this function is being used anywhere?

should it be subbed in here?

https://github.com/theanmolsharma/bcoin/blob/040d2519ba4e3f53ad014669aec64f5e8cdf451d/lib/wallet/coinselector.js#L239-L240